src/utils/styled_lines_iterator.rs

```diff
@@ -0,0 +1,224 @@
+use utils::LinesIterator;
+use style::*;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Generates rows of styled text in constrained width.
+///
+/// Given styled text and a width constraint, it iterates over
+/// collections of styled spans, each making up a row within the constraint.
+///
+/// # Examples
+///
+/// ```
+/// # use cursive::style::{StyledString, StyledText};
+/// # use cursive::utils::StyledLinesIterator;
+/// # let text_styled: &[StyledString] = &[].as_ref();
+/// let text_plain = text_styled.to_plain();
+/// let spans = StyledLinesIterator::make_spans(text_styled);
+/// // You can now keep `text_plain` and `spans` alive while using `iter`.
```

Comment:
Actually, we don't need `Cow` at all for `StyledSpan`, since it *always* points to the `StyledStr`.